### PR TITLE
CLOUDP-67051: Update CRD with TLS settings

### PR DESCRIPTION
**Pull request by fabianlindfors**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This PR adds TLS settings to the CRD. The design we settled upon looks like this:

```
apiVersion: mongodb.com/v1
kind: MongoDB
metadata:
  name: example-mongodb
spec:
  members: 3
  type: ReplicaSet
  version: "4.2.8"
  security:
    tls:
      enabled: true
      caConfigMapRef:
        name: "example-mongodb-ca"
      certificateKeySecretRef:
        name: "example-mongodb-tls"
```

### CODE REVIEW

File: deploy/crds/mongodb.com_mongodb_crd.yaml [hunk 1]
```diff
@@ -49,6 +49,46 @@ spec:
...
+                        The certificate is expected to be available under the key
+                        "ca.crt"
+                      properties:
+                        name:
+                          type: string
+                      required:
+                      - name
+                      type: object
+                    certificateKeySecretRef:
+                      description: CertificateKeySecret is a reference to a Secret
```
Comment (rodrigovalin) [inline deploy/crds/mongodb.com_mongodb_crd.yaml:72]: Maybe mention that this is a kubernetes.io/tls Secret type

File: pkg/apis/mongodb/v1/mongodb_types.go [hunk 2]
```diff
@@ -58,15 +58,23 @@ type TLS struct {
...
-	// CAConfigMapName is the name of a ConfigMap containing the certificate for the CA which signed the server certificates
+	// CaConfigMap is a reference to a ConfigMap containing the certificate for the CA which signed the server certificates
 	// The certificate is expected to be available under the key "ca.crt"
 	// +optional
-	CAConfigMapName string `json:"caConfigMapName"`
+	CaConfigMap LocalObjectReference `json:"caConfigMapRef"`
+}
+
+// LocalObjectReference is a reference to another Kubernetes object by name.
+// TODO: Replace with a type from the K8s API. CoreV1 has an equivalent
```
Comment (rodrigovalin) [inline pkg/apis/mongodb/v1/mongodb_types.go:73]: I assume then that this TODO won't show in our CRD, is that right?

Comment (fabianlindfors) [reply pkg/apis/mongodb/v1/mongodb_types.go:73]: Exactly! There TODO is on the `Name` field which the Operator SDK uses when generating the CRD.

File: deploy/crds/mongodb.com_mongodb_crd.yaml (hunk 1 above)
Comment (fabianlindfors) [reply deploy/crds/mongodb.com_mongodb_crd.yaml:72]: @rodrigovalin We don't require the secret to be of type `kubernetes.io/tls`, only that the right fields exist. Is it still worth mentioning?

Comment (rodrigovalin) [reply deploy/crds/mongodb.com_mongodb_crd.yaml:72]: it makes sense to mention that the secret is not expected to be "one type or another", that can't be enforced, as much as I understand, but it does not hurt to tell people that that's exactly the format we expect

File: deploy/crds/mongodb.com_mongodb_crd.yaml [hunk 3]
```diff
@@ -49,6 +49,46 @@ spec:
...
+                        containing a private key and certificate to use for TLS The
```
Comment (bznein) [inline deploy/crds/mongodb.com_mongodb_crd.yaml:73]: [nit] is a period missing between TSL and The?

File: deploy/crds/mongodb.com_mongodb_crd.yaml (hunk 1 above)
Comment (fabianlindfors) [reply deploy/crds/mongodb.com_mongodb_crd.yaml:72]: True! Updated the description now to mention this.

File: deploy/crds/mongodb.com_mongodb_crd.yaml (hunk 3 above)
Comment (fabianlindfors) [reply deploy/crds/mongodb.com_mongodb_crd.yaml]: Sure is. Good catch!

**Pull request merged**